### PR TITLE
Restructure Timimg Module as sdk and module

### PR DESCRIPTION
**Pull request by pprabhu-ng**
1. Add Folly based Timer sdk module under rns::sdk namespace with below interfaces

- Timer(double duration,bool repeats,callbackfunc,boot autostart);
- Timer::start()
- Timer::reschedule(duration,repeats)
- Timer::abort()
- Timer::getTimeRemaining()
- Timer::getFutureTime()

2. Update RSkTimingModule to use sdk timer module

### CODE REVIEW

File: ReactSkia/sdk/FollyTimer.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  }
+
+  if(timerCallback_.isScheduled()) {
+    RNS_LOG_WARN("Timer is already been scheduled for duration:" << targetDuration_);
+    return;
+  }
+
+  auto timeoutCb = std::bind(&Timer::scheduleTimerTimeout, this);
+  // For super fast, on-off timers, just enqueue them immediately rather than waiting
+  if(targetDuration_ <= 1) {
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/FollyTimer.cpp]: condition check to be less than 1ms ( similar to IOS)

File: ReactSkia/sdk/FollyTimer.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  if(cb_)
+    cb_();
+
+  if(repeats_) {
+     duration<double, std::milli> elapsed = system_clock::now() - timeoutFiredTime;
+     double schedulingOverhead = std::max(elapsed.count(), 0.0);
+     double targetDuration = std::max((targetDuration_ - schedulingOverhead), 0.0);
+
+     RNS_LOG_INFO("["<< this <<"] Reschedule repeat timer for duration:" << targetDuration);
+     if(targetDuration_ <= 1) {
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/FollyTimer.cpp]: local variable targetDuration and less than 1 check

File: ReactSkia/sdk/FollyTimer.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+void Timer::reschedule(double duration, bool repeats) {
+  RNS_LOG_DEBUG("["<< this << "] Reschedule timer for duration:" << duration << " repeats:" << repeats);
+
+  if(cb_ == nullptr) {
+    RNS_LOG_ERROR("No callback registered with timer, ignore scheduling of timer");
+    return;
+  }
+
+  repeats_ = repeats;
+  if((duration == targetDuration_) && timerCallback_.isScheduled()) {
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/FollyTimer.cpp]: this check not required in the timer. 
reschedule need to reset and reschedule with the duration called by client

File: ReactSkia/sdk/FollyTimer.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    RNS_LOG_WARN("Timer is already scheduled for this duration:" << targetDuration_);
+    return;
+  }
+
+  targetDuration_ = duration;
+
+  if(timerCallback_.cb == nullptr)
+     timerCallback_.cb = std::bind(&Timer::scheduleTimerTimeout, this);
+
```
Comment (pprabhu-ng) [inline ReactSkia/sdk/FollyTimer.cpp:88]: if already scheduled and target duration < 1 , cancel previous timeout and fire immediately.

File: ReactSkia/core_modules/RSkTimingModule.cpp [hunk 5]
```diff
@@ -76,73 +68,56 @@ void RSkTimingModule::createTimerForNextFrame(
...
+      auto it = lower_bound(jsTimersToCall.begin(), jsTimersToCall.end(), timer.second,
       [](auto& inList, auto& newItem ) { return (inList->target_ < newItem->target_);});
 
-      timersToCall.insert(it, timer.second.get());
-      RNS_LOG_TRACE("Expired TimerID=" << timer.second->callbackId_ << ", repeat=" << timer.second->repeats_ << ", duration=" << timer.second->duration_);
+      jsTimersToCall.insert(it, timer.second.get());
+      RNS_LOG_DEBUG("Expired TimerID=" << timer.second->callbackId_ << ", repeat=" << timer.second->repeats_ << ", duration=" << timer.second->duration_);
     } else {
-      RNS_LOG_TRACE("Pending TimerID=" << timer.second->callbackId_ << ", repeat=" << timer.second->repeats_ << ", duration=" << timer.second->duration_);
+      RNS_LOG_DEBUG("Pending TimerID=" << timer.second->callbackId_ << ", repeat=" << timer.second->repeats_ << ", duration=" << timer.second->duration_);
```
Comment (pprabhu-ng) [inline ReactSkia/core_modules/RSkTimingModule.cpp]: revert back to trace

**Review by munezbn — APPROVED** [review]

**Review by chatrapathi06 — APPROVED** [review]

**Pull request merged**